### PR TITLE
[frontend/e2e] End to end tests to cover all navigations on reports + background task execution from filters or search (#7378)

### PR DESCRIPTION
**Pull request by aHenryJard**
### Proposed changes

* End to end test coverage on report navigation
* End to end test coverage on background tasks from select all after filter and select all after search

### Related issues

* #7378 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case (coverage and e2e)
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### CODE REVIEW

File: opencti-platform/opencti-front/src/components/SearchInput.jsx [hunk 1]
```diff
@@ -103,12 +103,16 @@ const SearchInput = (props) => {
       size="small"
       placeholder={placeholder}
       onChange={(event) => {
+        console.log('ANGIE - onChange');
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/components/SearchInput.jsx]:
```suggestion
```

File: opencti-platform/opencti-front/src/components/SearchInput.jsx [hunk 2]
```diff
@@ -103,12 +103,16 @@ const SearchInput = (props) => {
...
         const { value } = event.target;
         setSearchValue(value);
       }}
       onKeyDown={(event) => {
+        console.log(`ANGIE - key down ${event.key}`);
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/components/SearchInput.jsx]:
```suggestion
```

File: opencti-platform/opencti-front/src/components/SearchInput.jsx [hunk 3]
```diff
@@ -103,12 +103,16 @@ const SearchInput = (props) => {
...
         const { value } = event.target;
+        console.log(`ANGIE - typeof onSubmit ${typeof onSubmit === 'function'}`);
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/components/SearchInput.jsx]:
```suggestion
```

File: opencti-platform/opencti-front/src/components/SearchInput.jsx [hunk 4]
```diff
@@ -103,12 +103,16 @@ const SearchInput = (props) => {
...
         if (typeof onSubmit === 'function' && event.key === 'Enter') {
+          console.log('ANGIE - SUBMIT');
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/components/SearchInput.jsx]:
```suggestion
```

File: opencti-platform/opencti-front/tests_e2e/backgroundTask/backgroundTask.spec.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    await sleep(30000);
+    await tasksPage.goto();
+    await expect(tasksPage.getPage()).toBeVisible();
+    const isTaskProcessing2 = await page.getByText('Waiting').first().isVisible({ timeout: 200 });
+    if (isTaskProcessing2) {
+      await sleep(30000);
+      await tasksPage.goto();
+      await expect(tasksPage.getPage()).toBeVisible();
+    }
+  }
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/tests_e2e/backgroundTask/backgroundTask.spec.ts]: ah I forgot this part 😅 -- I need to change that sorry.

File: opencti-platform/opencti-front/src/private/components/events/incidents/IncidentsLines.tsx [hunk 6]
```diff
@@ -105,24 +105,26 @@ const IncidentsLines: FunctionComponent<IncidentsLinesProps> = ({
...
-      dataColumns={dataColumns}
-      nbOfRowsToLoad={nbOfRowsToLoad}
-      onLabelClick={onLabelClick}
-      paginationOptions={paginationOptions}
-      selectedElements={selectedElements}
-      deSelectedElements={deSelectedElements}
-      selectAll={selectAll}
-      onToggleEntity={onToggleEntity}
-    />
+    <div data-testid='incident-page'>
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/events/incidents/IncidentsLines.tsx]: Why the test-id is on the Lines component and not `Incidents.tsx` ?

File: opencti-platform/opencti-front/tests_e2e/model/report.pageModel.ts [hunk 7]
```diff
@@ -1,8 +1,13 @@
 import { Page } from '@playwright/test';
 
 export default class ReportPage {
+  pageUrl = '/dashboard/analyses/reports';
   constructor(private page: Page) {}
 
+  async goto() {
+    await this.page.goto(this.pageUrl);
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/tests_e2e/model/report.pageModel.ts:8]: As far as possible, we should avoid using the goto function as it reloads the page (if I'm right) and slow the test, instead we should use the left navigation menu

File: opencti-platform/opencti-front/tests_e2e/model/reportDetails.pageModel.ts [hunk 8]
```diff
@@ -18,8 +19,13 @@ export default class ReportDetailsPage {
     return this.page.getByLabel('Update', { exact: true });
   }
 
-  goToOverviewTab() {
-    return this.page.getByRole('tab', { name: 'Overview' }).click();
+  async goToOverviewTab() {
+    await this.page.getByRole('tab', { name: 'Overview' }).click();
+    await expect(this.page.getByTestId('report-overview')).toBeVisible();
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/tests_e2e/model/reportDetails.pageModel.ts]: I think this expect should be done in the test and not the page model to not mix concerns. Page model to help manipulating the app, tests for assertions. What do you think?

Comment (aHenryJard) [reply opencti-platform/opencti-front/tests_e2e/model/reportDetails.pageModel.ts]: I don't have strong opinion on that it's fine for me, I'm moving.

File: opencti-platform/opencti-front/tests_e2e/model/report.pageModel.ts (hunk 7 above)
Comment (aHenryJard) [reply opencti-platform/opencti-front/tests_e2e/model/report.pageModel.ts:8]: As far as I see it's often use for starting tests, I just replaced existing calls. It's true that it's taking time. i prefer having the page URL at only one place that is why I moved it.

File: opencti-platform/opencti-front/src/private/components/events/incidents/Incident.tsx [hunk 9]
```diff
@@ -99,6 +99,7 @@ const Incident = ({
         container={true}
         spacing={3}
         classes={{ container: classes.gridContainer }}
+        data-testid='incident-page'
```
Comment (aHenryJard) [inline opencti-platform/opencti-front/src/private/components/events/incidents/Incident.tsx]:
```suggestion
```

**Review by lndrtrbn — APPROVED** [review]

**Pull request merged**